docs: Document scan-sessions (#217)
Fraunhofer-AISEC/gallia: docs/uds/scan_modes.md

From a high level perspective, the destination id is iterated and a valid payload is sent.
If a valid answer is received, an ECU has been found.

## Session Scan

UDS has the concept of sessions.
Different sessions can for example offer different services.
A session is identified by a 1 byte session ID.
The UDS standard defines a set of well known session IDs, but vendors are free to add their own sessions.
Some sessions might only be available from a specific ECU state (e.g. current session, enabled/configured ECU features, coding, ...).
Most of those preconditions cannot be detected automatically and might require vendor specific knowledge.

> This scanner tries to find all available session transitions.

Reviewer: ```suggestion
The session scan tries to find all available session transitions.
```

> Starting from the default session (0x01), it iterates all session IDs and tries to access this session.

Reviewer: ```suggestion
Starting from the default session (0x01), all session IDs are iterated and enabling the relevant session is tried.
```

> If the ECU replays with a positive response, the session is available, else the session is not available (at leased from the current state).

Reviewer: ```suggestion
If the ECU replies with a positive response, the session is available.
In case of a negative response, the session is considered not available from the current state.
```

> To detect sessions, which are only reachable from a session different to the default session,

Reviewer: ```suggestion
To detect sessions, which are only reachable from a session different to the default session, a recursive approach is used.
The scan for new sessions starts at each previously identified session.
The maximum depth is limited to avoid endless scans in case of transition cycles, such as `0x01 -> 0x03 -> 0x05 -> 0x03`.
```

> this scanner uses a recursive approach and starts scanning for new sessions at every previously identified session.

Reviewer: ```suggestion
```
